### PR TITLE
Abstract action lambda

### PR DESCRIPTION
**Pull request by DanVanAtta**
34c1f43 adds infrastructure code so we can replace "new AbstractAction() { }" with a lambda expression (refactor per #410)

The next commits does some bulk conversions from one construct to the other. Opportunistically I've also marked a number of methods as static as well.

The last commit backs out changes to one file that was getting a compiler error in Travis only: http://travis-ci.org/triplea-game/triplea/builds/107405379

[Review on Reviewable](https://reviewable.io/reviews/triplea-game/triplea/439)

### CODE REVIEW

**Comment by gaborbernat** [comment]
Review status: 0 of 62 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/games/strategy/common/ui/BasicGameMenuBar.java, line 751 \[r1\]](https://reviewable.io:443/reviews/triplea-game/triplea/439#-KAOpN2CrUOL4L83AHom:-KAOpN2CrUOL4L83AHon:-883611877) ([raw file](https://github.com/triplea-game/triplea/blob/e1e82754a2c35dc5344886a22f2c9ef8e09d6a67/src/games/strategy/common/ui/BasicGameMenuBar.java#L751)):*
this seems like a new line added. what's this?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/triplea-game/triplea/439)*

**Comment by DanVanAtta** [comment]
Review status: 0 of 62 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/games/strategy/common/ui/BasicGameMenuBar.java, line 751 \[r1\]](https://reviewable.io:443/reviews/triplea-game/triplea/439#-KAOpN2CrUOL4L83AHom:-KAP18QKsdahnUrernTe:1248711108) ([raw file](https://github.com/triplea-game/triplea/blob/e1e82754a2c35dc5344886a22f2c9ef8e09d6a67/src/games/strategy/common/ui/BasicGameMenuBar.java#L751)):*
Here the problem is we had a helper function defined as part of the anonymous AbstractAction class. This helper function 'exportXMLFile()' was moved up to the parent class. It is a bit hard to tell, but there are actually only two lines in the addExportXML file method, and in the before and after case, note the last line in the method is both times: " parentMenu.add(exportXML).setMnemonic(KeyEvent.VK_X);"

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/triplea-game/triplea/439)*

**Comment by gaborbernat** [comment]
Review status: 0 of 62 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/games/strategy/common/ui/BasicGameMenuBar.java, line 751 \[r1\]](https://reviewable.io:443/reviews/triplea-game/triplea/439#-KAOpN2CrUOL4L83AHom:-KAPFbUbxvDuQeM9_5Eg:-1775285248) ([raw file](https://github.com/triplea-game/triplea/blob/e1e82754a2c35dc5344886a22f2c9ef8e09d6a67/src/games/strategy/common/ui/BasicGameMenuBar.java#L751)):*
I see thanks for clearing that up.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/triplea-game/triplea/439)*

**Comment by DanVanAtta** [comment]
Now rebased on branch engine_version (preceding PR) with merge conflicts resolved

**Comment by ron-murhammer** [comment]
@DanVanAtta Looks like this needs rebased.

**Comment by DanVanAtta** [comment]
conflicts resolved, now rebased

File: src/games/strategy/common/swing/SwingAction.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ */
+public class SwingAction {
+
+  /**
+   * @param name Name for the abstract action, passed along to the AbstractAction constructor.
+   * @param swingAction Lambda java.util.function.Consumer object, accepts one arg and returns void.
+   */
+  public static AbstractAction of(final String name, final Consumer<ActionEvent> swingAction) {
+    return new AbstractAction(name) {
+      private static final long serialVersionUID = 1L;
```
Comment (ron-murhammer) [inline src/games/strategy/common/swing/SwingAction.java]: So looks like all AbstractAction's will now have the serialVersionUID of 1 where they all had different/random ones before. Is this gonna cause any compatibility issues?

**Comment by ron-murhammer** [comment]
@DanVanAtta - Really like this refactor. Just 1 question above around serialVersionUID.

File: src/games/strategy/common/swing/SwingAction.java (hunk 1 above)
Comment (DanVanAtta) [inline src/games/strategy/common/swing/SwingAction.java]: Pushed: 44041f9 ,  replaced the "1L" with a generated  random UID.

I had to refresh a bit on serial version UID:
http://stackoverflow.com/questions/285793/what-is-a-serialversionuid-and-why-should-i-use-it

I'm about 95% sure the answer is the different serial version UID does not matter.  It's hard to be certain if we rely on the deserialization of these objects. If we do, then at least we'll have reduced the risk of future unnecessary cross-version ClassCastExceptions.

**Comment by ron-murhammer** [comment]
@DanVanAtta - Alright. Gonna merge this now.

**Pull request merged**